model/flow/transaction.go

```diff
@@ -245,15 +267,37 @@ func (tb *TransactionBody) SignPayload(address Address, keyID uint64, privateKey
 	return nil
 }
 
-// SignEnvelope signs the full transaction (payload + payload signatures) with the specified account key.
+// SignEnvelope signs the full transaction (payload + payload signatures) with the specified account key using the default transaction domain tag.
 //
 // The resulting signature is combined with the account address and key ID before
 // being added to the transaction.
 //
 // This function returns an error if the signature cannot be generated.
-func (tb *TransactionBody) SignEnvelope(address Address, keyID uint64, privateKey crypto.PrivateKey, hasher hash.Hasher) error {
+func (tb *TransactionBody) SignEnvelope(
+	address Address,
+	keyID uint64,
+	privateKey crypto.PrivateKey,
+	hasher hash.Hasher,
+) error {
+	return tb.SignEnvelopeWithTag(address, keyID, privateKey, hasher, TransactionDomainTag[:])
+}
 
-	sig, err := privateKey.Sign(tb.EnvelopeMessage(), hasher)
+// SignEnvelope signs the full transaction (payload + payload signatures) with the specified account key.
+//
+// The resulting signature is combined with the account address and key ID before
+// being added to the transaction.
+//
+// This function returns an error if the signature cannot be generated.
+func (tb *TransactionBody) SignEnvelopeWithTag(
+	address Address,
+	keyID uint64,
+	privateKey crypto.PrivateKey,
+	hasher hash.Hasher,
+	tag []byte,
+) error {
+	message := tb.EnvelopeMessage()
+	message = append(tag[:], message...)
+	sig, err := privateKey.Sign(message, hasher)
```

Comment:
I think it would be clearer to have a single function `signWithTag` that takes a message and a tag, prepends the message and calls `Sign`. 
Then `signWithTag` would be called by both `SignEnvelope` and `SignPayload` using different messages. 
The point is to centralize how `message` and `tag` mix in the same function.

Reply:
applied! Thank you for the suggestion.